satpy/readers/slstr_l2.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+""" Reader for Sentinel-3 SLSTR SST data
+"""
+
+from datetime import datetime
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy import CHUNK_SIZE
+import xarray as xr
+
+
+class SLSTRL2FileHandler(BaseFileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info, engine=None):
+        super(SLSTRL2FileHandler, self).__init__(filename, filename_info, filetype_info)
+
+        if filename.endswith('tar'):
+            import tarfile
+            import os
+            import tempfile
+            with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
Is this directory deleted properly when satpy is finished ?

Reply:
Normally since its wrapped in "with" it should remove the tempdir when out of scope.  I did a small check checked and it does indeed remove the directory.

Reply:
Perfect